File: type_cast_error/android/app/src/main/kotlin/dev/yanshouwang/type_cast_error/MyApi.kt
package dev.yanshouwang.type_cast_error

import android.util.Log

class MyApi : MyHostApi {
    companion object {
        val TAG: String = MyApi::class.java.simpleName
    }

    override fun testTypeCastError(testArgs: MyTestArgs) {
        val itemsArgs = testArgs.itemsArgs
        for (itemArgs in itemsArgs) {
            val key = itemArgs.key
            val value = itemArgs.value
            Log.d(TAG, "testTypeCastError: $key: $value")
        }
    }
}

File: type_cast_error/my_api.dart
import 'package:pigeon/pigeon.dart';

@ConfigurePigeon(
  PigeonOptions(
    dartOut: 'lib/my_api.g.dart',
    dartOptions: DartOptions(),
    kotlinOut:
        'android/app/src/main/kotlin/dev/yanshouwang/type_cast_error/MyApi.g.kt',
    kotlinOptions: KotlinOptions(
      package: 'dev.yanshouwang.type_cast_error',
    ),
  ),
)
@HostApi()
abstract class MyHostApi {
  void testTypeCastError(MyTestArgs testArgs);
}

class MyTestArgs {
  Map<int?, Uint8List?> itemsArgs;

  MyTestArgs(this.itemsArgs);
}
